components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -756,13 +756,109 @@ IJsonSerializer jsonSerializer
             PublishedNodesEntryModel request,
             CancellationToken ct) {
             _logger.Information("{nameof} method triggered", nameof(UnpublishAllNodesAsync));
+            var sw = Stopwatch.StartNew();
             await _lockConfig.WaitAsync(ct).ConfigureAwait(false);
+            var response = new List<string>();
             try {
-                throw new MethodCallStatusException((int)HttpStatusCode.NotImplemented, "Not Implemented");
+
+                var found = false;
+                // Perform pass to determine existing groups
+                var matchingGroups = new List<PublishedNodesEntryModel>();
+                foreach (var entry in _publishedNodesEntries) {
+                    if (entry.HasSameGroup(request)) {
+                        // We may have several entries with the same DataSetGroup definition,
+                        // so we will remove nodes only if the whole DataSet definition matches.
+                        if (IsSameDataSet(entry, request)) {
+                            entry.OpcNodes.Clear();
+                            found = true;
+                        }
+                        matchingGroups.Add(entry);
+                    }
+                }
+
+                // Report error if no matching endpoint was found.
+                if (matchingGroups.Count == 0) {
+                    throw new MethodCallStatusException((int)HttpStatusCode.NotFound,
+                        $"Endpoint not found: {request.EndpointUrl}");
+                }
+
+                // Report error if there were entries that did not have any nodes
+                if (!found) {
+                    var entriesNotFoundJson = _jsonSerializer.SerializeToString(request);
```

Comment:
Does it bring much value to report the request itself in the response here ? I would suggest to just throw an endpoint not found exception similar to the one above and even remove the check above for `matchingGroups.Count == 0` and only throw based on whether we found exact data set or not.

Reply:
Indeed this check could be omitted.